x-pack/plugins/observability_solution/inventory/public/hooks/use_discover_redirect.ts

```diff
@@ -4,74 +4,53 @@
  * 2.0; you may not use this file except in compliance with the Elastic License
  * 2.0.
  */
-import {
-  ENTITY_DEFINITION_ID,
-  ENTITY_DISPLAY_NAME,
-  ENTITY_LAST_SEEN,
-  ENTITY_TYPE,
-} from '@kbn/observability-shared-plugin/common';
-import { useCallback } from 'react';
+import { useCallback, useMemo } from 'react';
 import type { InventoryEntity } from '../../common/entities';
 import { useKibana } from './use_kibana';
-import { useUnifiedSearchContext } from './use_unified_search_context';
+import { useEntityDefinition } from './use_entity_definition';
+import { useAdHocDataView } from './use_adhoc_data_view';
 
-const ACTIVE_COLUMNS = [ENTITY_DISPLAY_NAME, ENTITY_TYPE, ENTITY_LAST_SEEN];
-
-export const useDiscoverRedirect = () => {
+export const useDiscoverRedirect = (entity: InventoryEntity) => {
   const {
     services: { share, application, entityManager },
   } = useKibana();
+  const { entityDefinitions } = useEntityDefinition(entity.entityDefinitionId);
```

Comment:
Can we rename it so it makes clear that this hooks fetch data? 
```suggestion
  const { entityDefinitions } = useFetchEntityDefinition(entity.entityDefinitionId);
```

Reply:
Renamed 👍